hexa/datasets/queue.py

```diff
@@ -1,18 +1,81 @@
 from logging import getLogger
 
+import pandas as pd
+from django.conf import settings
 from dpq.queue import AtLeastOnceQueue
 
-from hexa.datasets.models import DatasetFileMetadataJob
+from hexa.datasets.api import generate_download_url
+from hexa.datasets.models import (
+    DatasetFileMetadata,
+    DatasetFileMetadataJob,
+    DatasetVersionFile,
+)
 
 logger = getLogger(__name__)
 
 
+def read_file_content(download_url: str, content_type: str) -> pd.DataFrame:
+    try:
+        if content_type == "text/csv":
+            return pd.read_csv(download_url)
+        elif content_type == "application/octet-stream":
+            return pd.read_parquet(download_url)
+        else:
+            raise ValueError(f"Unsupported content type: {content_type}")
+    except pd.errors.ParserError as e:
+        print(f"Error parsing the file content: {e}")
+        return pd.DataFrame()
+    except ValueError as e:
+        print(f"Unsupported file content: {e}")
+        return pd.DataFrame()
+
+
 def create_dataset_file_metadata_task(
     queue: AtLeastOnceQueue, job: DatasetFileMetadataJob
 ):
-    # TODO: imlpement ticket PATHWAYS-98  - extract data in background task
-    dataset_version_file_id = job.args["fileId"]
-    logger.info(f"Creating dataset version file {dataset_version_file_id}")
+    try:
+        dataset_version_file_id = job.args["file_id"]
+        dataset_version_file = DatasetVersionFile.objects.get(
+            id=dataset_version_file_id
+        )
+        logger.info(
+            f"Creating dataset snapshot for version file {dataset_version_file_id}"
+        )
+        dataset_file_metadata = DatasetFileMetadata.objects.create(
+            dataset_version_file=dataset_version_file,
+            status=DatasetFileMetadata.STATUS_PROCESSING,
+        )
+
+        download_url = generate_download_url(dataset_version_file)
+        file_snapshot_df = read_file_content(
+            download_url, dataset_version_file.content_type
+        )
+        if not file_snapshot_df.empty:
+            file_snapshot_content = file_snapshot_df.head(
+                settings.WORKSPACE_DATASETS_FILE_SNAPSHOT_SIZE
+            )
+            dataset_file_metadata.content = file_snapshot_content.to_json(
+                orient="records"
+            )
+            logger.info(f"Dataset snapshot saved for file {dataset_version_file_id}")
+        else:
+            logger.info(f"Dataset snapshot is empty for file {dataset_version_file_id}")
+        dataset_file_metadata.status = DatasetFileMetadata.STATUS_FINISHED
+        dataset_file_metadata.save()
+        logger.info("Dataset snapshot created for file {dataset_version_file_id}")
+    except Exception as e:
```

Comment:
This whole block is to be reviewed. You do not know why it raised an error but you still try to get the DatasetVersionFile & the DatasetFileMetadata.

Reply:
You have to distinguish the `DatesetVersionFile.DoesNotExist` from the rest as a start

Reply:
Agree yeah I will rewrite together with covering edge cases in tests